llvm/unittests/IR/DataLayoutTest.cpp

```diff
@@ -312,12 +312,12 @@ TEST(DataLayout, ParsePointerSpec) {
         "p16777215:32:32:64:8", "p16777215:16777215:32768:32768:16777215"})
     EXPECT_THAT_EXPECTED(DataLayout::parse(Str), Succeeded());
 
-  for (StringRef Str :
-       {"p", "p0", "p:32", "p0:32", "p:32:32:32:32:32", "p0:32:32:32:32:32"})
+  for (StringRef Str : {"p", "p0", "p:32", "p0:32", "p:32:32:32:32:32:32",
+                        "p0:32:32:32:32:32:32"})
     EXPECT_THAT_EXPECTED(
         DataLayout::parse(Str),
         FailedWithMessage("malformed specification, must be of the form "
-                          "\"p[<n>]:<size>:<abi>[:<pref>[:<idx>]]\""));
+                          "\"p[<n>]:<size>:<abi>[:<pref>[:<idx>[:<addr>]]]\""));
```

Comment:
Not sure about this nesting, wouldn't it be easier to read as `p[<n>]:<size>:<abi>[:<pref>][:<idx>][:<addr>]`

Reply:
The point of this nesting is that you can only specify `idx` iff you also specify `pref`. I guess this applies to `addr` as well.

Reply:
I guess that makes sense. Should I update the LangRef one as well?